replace vue compiler for vue 3 compatibility

lzurbriggen opened the pull request:
I replaced the old `vue-template-compiler` with the new vue 3 `@vue/compiler-sfc` to prevent version mismatch errors when using this package with vue 3.

(also cleaned up eslint errors)

I'm not sure how you want to handle dependencies/releases. `compiler-sfc` is on `3.0.0-rc.5`, if you accept the change maybe this should be a `next` version until there's a stable `3.0.0` release.

vperron reviewed (approved):
Agreed ! With 3 small gotchas though.

- I think you're gonna have to rebase this as soon as #92  is merged :/ sorry ! but it should be quick.
- it would be probably way better to separate the style (eslint) updates in a different commit
- @vue/compiler-sfc now has a definitive v3.0.0 :)

vperron commented:
There you go @lzurbriggen :) A few small more steps :+1:

lzurbriggen commented:
@vperron I updated the compiler dependency and split code style changes to a second commit (indentation adjustments in the tests are mostly because of changes in the compiler output).

vperron commented on src/javascript-extract.js:
```diff
@@ -24,12 +24,15 @@ function getGettextEntriesFromJavaScript(argTokens = []) {
+      if (!DEFAULT_VUE_GETTEXT_FUNCTIONS.hasOwnProperty(gettextFunc)) {
+        continue;
+      }
+      let args = DEFAULT_VUE_GETTEXT_FUNCTIONS[gettextFunc];
```
I don't understand this change, can you maybe place a comment to explain why it's so ?

lzurbriggen replied:
I should have paid closer attention when rebasing. I tried to fix a problem (`guard-for-in` eslint violation) that has been solved in the `added babel parser support for javascript parsing` PR. I'll remove it.

vperron commented on src/extract.js:
```diff
@@ -574,7 +566,8 @@ exports.Extractor = class Extractor {
 
-  _extractTranslationData(filename, content) {
+  _extractTranslationData(filename, filecontent) {
+    let content = filecontent;
```
I don't understand why this whole renaming of `content` to `filecontent` is necessary, especially if you rename it to `content` at the very end ^^ 

Either finish the job and rename everything to `fileContent` in a separate commit (paying attention to the camelCase convention that exists in this package) or don't do it ?

lzurbriggen replied:
I honestly didn't remember (it's been a while). The original code violates the `no-param-reassign` (and `no-shadow` on line 276) eslint rule.

I'll rename it to fileContent.

vperron replied:
Oh, that's interesting.

vperron commented:
That is fantastic ! I paid attention to do a second review, and seen a few little gotchas. ONce you address them, I'll gladly merge this immediately :partying_face:

vperron commented:
Thanks for your contributions @lzurbriggen  :) I made you a write-approved contributor to this repository !
